java/code/src/com/redhat/rhn/domain/access/WebEndpoint.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2025 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.redhat.rhn.domain.access;
+
+import com.redhat.rhn.domain.BaseDomainHelper;
+
+import java.util.Objects;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.NamedNativeQuery;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "endpoint", schema = "access")
+@NamedNativeQuery(name = "WebEndpoint_user_access_endpoint_scope",
+        // Select an accessible endpoint by path, http method (GET, POST, etc.) and scope (Web UI or API)
+        query = "SELECT * FROM access.userAccessTable e WHERE user_id = :user_id AND e.endpoint = :endpoint " +
+                "AND e.http_method = :http_method AND e.scope = :scope LIMIT 1"
+)
+@NamedNativeQuery(name = "WebEndpoint_user_access_endpoint",
+        // Select an accessible endpoint by path, http method (GET, POST, etc.)
+        query = "SELECT * FROM access.userAccessTable e WHERE user_id = :user_id AND e.endpoint = :endpoint " +
+                "AND e.http_method = :http_method LIMIT 1"
+)
+@NamedNativeQuery(name = "WebEndpoint_user_access_class_method",
+        // Select an accessible endpoint by its handler class and method
+        query = "SELECT * FROM access.userAccessTable e WHERE user_id = :user_id AND e.scope = :scope " +
+                "AND e.class_method = :class_method LIMIT 1"
+)
+@NamedNativeQuery(name = "WebEndpoint_get_unauthenticated_webui",
+        // Get unauthenticated Web UI endpoints
+        query = "SELECT endpoint FROM access.endpoint WHERE auth_required = false"
```

Comment:
The next query is filtering by API, this one is only filtering by auth_required false. Do we need the two methods? Is this needed because the default handler for the XML-RPC API?

Reply:
This one is actually used with HTTP API routes as well (works the same way as a web UI endpoint). I'll fix the naming to reflect that.